chalk-solve/src/coherence/orphan.rs

```diff
@@ -13,14 +13,11 @@ use failure::Fallible;
 //     forall<T> { LocalImplAllowed(MyType<T>: Trait) }
 //
 // This must be provable in order to pass the orphan check.
-pub fn perform_orphan_check<DB>(
-    db: &DB,
+pub fn perform_orphan_check(
+    db: &ChalkSolveDatabase,
```

Comment:
```suggestion
    db: &dyn ChalkSolveDatabase,
```

Reply:
Might be a good idea and E-good-first-bug to add this check to CI

Reply:
I suppose we should just add `#[deny(rust_2018_idioms)]` or whatever it is